python/semantic_kernel/kernel.py

```diff
@@ -398,7 +399,12 @@ async def invoke_function_call(
         frc = FunctionResultContent.from_function_call_content_and_result(
             function_call_content=function_call, result=invocation_context.function_result
         )
-        chat_history.add_message(message=frc.to_chat_message_content())
+
+        is_streaming = any(isinstance(message, StreamingChatMessageContent) for message in chat_history.messages)
```

Comment:
There are two more places in this method where you'd need to check if the chat is carried out in a streaming manner.

Reply:
I don’t follow?

Reply:
Are you referring to these: `message=frc.to_chat_message_content()`? I don't think it matters that those "error" messages are converted to the Streaming CMC. The fact that we have a Streaming CMC as a response from the model, should indicate enough that we are streaming -- I was doing this check to avoid having to introduce a new `is_streaming` kwarg to the invoke_function_call method.

Reply:
Now that I think about it, too: if we converted those intermittent "error" messages to Streaming CMC, then if someone does combine the streaming chat message chunks together, they would see those "hey model, please reformat the json and try calling again" messages...

Reply:
Another question is: when do we need to concatenate the tool result messages?

Reply:
We have examples of concatenating other messages, so this would be an example on how to do that.